GVFS/GVFS.Upgrader/Program.cs

```diff
@@ -9,8 +9,18 @@ public static void Main(string[] args)
         {
             GVFSPlatformLoader.Initialize();
 
-            Parser.Default.ParseArguments<UpgradeOrchestrator>(args)
+            UpgradeOrchestrator upgraderType = UpgradeOrchestratorFactory.Create();
+
+            if (upgraderType is MacUpgradeOrchestrator)

```

Comment:
This whole chunk is confusing to me.

1. We should use our base type to avoid this `is` convention.
2. We create `upgraderType` just to check the `is`, and then we don't even use it.

It would be better if the `UpgradeOrchestrator` had some `ExecuteOnArgs(string [] args)` method that did this internal logic according to its own type.

Reply:
I see now that there is no `WindowsUpgradeOrchestrator`, which would probably be helpful here.

Reply:
I was able to clean this up a bit. I now have a stable `UpgradeOptions` class that then gets passed to a specific `UpgradeOrchestrator` implementation.